addons/survey/static/src/js/survey_form.js

```diff
@@ -253,15 +254,30 @@ publicWidget.registry.SurveyFormWidget = publicWidget.Widget.extend(SurveyPreloa
         }
     },
 
-    _onSubmit: function (event) {
+    async _onSubmit(event) {
         event.preventDefault();
         const options = {};
         const target = event.currentTarget;
         if (target.value === 'previous') {
             options.previousPageId = parseInt(target.dataset['previousPageId']);
         } else if (target.value === 'next_skipped') {
             options.nextSkipped = true;
+        } else if (target.value === 'finish' && ['btn btn-primary', 'btn btn-secondary'].includes(target.classList.value)) {
+            options.isFinish = true;
         } else if (target.value === 'finish') {
+            // Adding pop-up only when the test is submitted using next button
+            const confirmed = await new Promise((resolve) => {
+                this.call("dialog", "add", ConfirmationDialog, {
+                    title: "Submit confirmation",
+                    body: "Are you sure you want to submit the survey?",
+                    confirmLabel: "Submit",
+                    confirm: () => resolve(true),
+                    cancel: () => resolve(false),
+                });
+            });
+            if (!confirmed) {
+                return;
+            }
```

Comment:
i don't think we need Promise over here
we can achieve it without using promise